aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp

```diff
@@ -30,20 +30,32 @@ static void addcmul_cpu_kernel(TensorIterator& iter, Scalar value) {
 
 static void addcdiv_cpu_kernel(TensorIterator& iter, Scalar value) {
   ScalarType dtype = iter.dtype(0);
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX(dtype, "addcdiv_cpu_out", [&] {
-    scalar_t scalar_val = value.to<scalar_t>();
-    auto scalar_vec = Vec256<scalar_t>(scalar_val);
-    cpu_kernel_vec(
-        iter,
-        [=](scalar_t self_val, scalar_t t1_val, scalar_t t2_val) -> scalar_t {
-          return self_val + scalar_val * t1_val / t2_val;
-        },
```

Comment:
Do all integral types go through this non-vectorised version? Otherwise, I would be concerned about possible performance regression.

Reply:
In my opinion all integral types go through this branch. Or did I miss something?

Reply:
I have no idea. Could you please run some benchmarks?

Reply:
I did some tests based the original code.  The follows dtypies go through this non-vectorised version.
**[torch.uint8, torch.int8, torch.int16 or torch.short, torch.int32 or torch.int and torch.int64 or torch.long]**
So I think there should not be performance regression. If necessary, I'll do some benchmarks.

Reply:
TBH, I'm out of my depth here. Perhaps @andreaskoepf or @VitalyFedyunin can look at this. I don't know whether the operation on integral types is never vectorised, regardless of platform.

Reply:
Much thanks for your advices !!